### PR TITLE
Refactor tracing/audit plumbing: RequestAudit seam + retire RawBody debt

### PR DESCRIPTION
**Pull request by hooyao**
Two coupled, review-driven refactors of the per-request audit/tracing path. **No user-facing behavior change** — audit artifact bytes, trace-file names, and wire bytes are all unchanged; the win is a single gating seam, honest types, and an allocation-free off-trace hot path.

## 1. `RequestAudit` seam (P1/P2/P3)

The concept *"are we tracing this request?"* was expressed five ways across six files, plus two **unguarded** sites that did audit-only work (re-serializing the request body, stashing unconditionally) even when tracing is off.

- **New `Pipeline/RequestAudit.cs`** — per-request scoped service owning the one `Enabled` flag + the four `Record*` emissions (no-op when off) + the trace-only buffer factories. Replaces the scattered `_tracingEnabled` fields / endpoint locals / `?:` ternaries.
- **Both strategies stash the exact POSTed bytes** on `Response.UpstreamWireBody` (gated on `audit.Enabled`); endpoints audit those instead of re-serializing the IR. Kills a multi-MB redundant serialize on the off-trace hot path, and makes `upstream-req` the *exact* wire bytes rather than a re-derivation.
- **`UpstreamWireBody` null-overload collapsed** — it now means only "captured wire bytes"; backend identity is read from `Target.Vendor`.
- All four audit-emitting endpoints (`messages`/`codex`/`count_tokens`/`models`) route through the seam — zero direct-logger audit sites remain.

## 2. Retire `RawBody` + inbound-buffer debt

`BridgeRequest.RawBody` was a wire-layer field mislayered onto pipeline-internal state — the pipeline never read its bytes (only `.Length`, for one debug line), its doc-comment was stale, and it forced the pooled read buffer to stay alive across the whole request. The inbound length is also stale the moment the first body-mutating stage runs (even cc→opus), so it isn't meaningful pipeline state.

- **Delete `RawBody`**; log the inbound size at the endpoint boundary where it's observed.
- **New `Endpoints/InboundBody.cs`** — `ReadPooledAsync` returns a disposable `PooledBody` (ArrayPool + doubling grow, idempotent `Dispose`, capacity-vs-length encapsulated), replacing three copy-pasted `ReadBodyPooledAsync` methods. The body is read in a synchronous-scope `using` so the pooled buffer no longer crosses `await`.
- **Delete dead `bodyPooled` plumbing** (logger param, `BridgeIoPayload.BodyPooled` + `Release()`, sink worker's `Release()` call) — every caller passed `false`, so the pool-return path was unreachable.

## Testing

- **564 unit tests pass** (0 warnings). New from-contract edge-case suites: `RequestAuditSeamTests`, `RequestAuditCodexSeamTests`, `InboundBodyTests` — each mutation-checked (break product → RED → restore).
- Key invariants asserted: zero-overhead-when-off (no audit events, no wire-body stash), `upstream-req` == exact POSTed bytes, each artifact is the wire on its boundary (never the internal IR), pooled-buffer dispose-safety.
- **AOT publish clean** — no new dependency, `.exe` size unchanged (~12 MB), no IL/trim warnings.
- Reviewed with the code-reviewer agent (no correctness/AOT/lifetime defects); its three minor notes (a tracing-only double-copy in count_tokens, a dead marker class, two stray usings) were all fixed.

Net **−100 lines** of source despite adding the seam + helper.

OpenSpec: two changes (`refactor-tracing-audit-seam`, `retire-rawbody-and-inbound-buffer-debt`) synced to `observability` (+3 reqs) and `pipeline-request-isolation` (+1 req), then archived.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/CopilotBridge.Cli/Endpoints/InboundBody.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                buffer.CopyTo(buf);
+                reader.AdvanceTo(buffer.End);
+                return new PooledBody(buf, len);
+            }
+
+            // Not complete: consumed nothing, examined everything — the pipe keeps
+            // accumulating the rest while retaining what we have already seen, so the
+            // final (IsCompleted) read returns the whole body in one sequence.
+            reader.AdvanceTo(buffer.Start, buffer.End);
+        }
```
[the remark this answers is not in the document]
Comment (hooyao) [reply src/CopilotBridge.Cli/Endpoints/InboundBody.cs]: You're right — this was a real production-hanging regression. Confirmed against the official System.IO.Pipelines docs, which list this exact pattern (examine-without-consume until IsCompleted) as a ❌ OOM/stall anti-pattern, and against Kestrel's ~1 MB MaxRequestBufferSize backpressure. Since Claude Code bodies routinely exceed 4 MB it would have hung the /cc, /codex, and count_tokens paths.

Fixed by reverting off PipeReader entirely and reading Request.Body (a Stream), which consumes as it reads and cannot deadlock. The read is now backed by Microsoft.IO.RecyclableMemoryStream (pooled chunks, no hand-rolled growth) rather than the old ArrayPool loop. A code comment at the helper documents why it must stay a Stream, not a PipeReader, so it isn't reintroduced.

Also worth noting: this was invisible to the unit tests because a MemoryStream-backed reader has no backpressure — so the guard is now the Stream API choice itself, documented, rather than a test. Two follow-up adversarial reviews of the RecyclableMemoryStream version caught additional issues (buffer pinned for the whole request; unbounded pool free-list; a pre-existing hardcoded count_tokens audit URL), all fixed in the latest push. Thanks for the catch.
